packages/react/src/dialog/Dialog.tsx

```diff
@@ -0,0 +1,50 @@
+import * as RadixDialog from "@radix-ui/react-dialog";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { CloseIcon } from "./CloseIcon";
+import * as styles from "./Dialog.css";
+
+type DialogProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  ComponentPropsWithRef<typeof RadixDialog.Root>,
+  {
+    children: ReactNode;
+    hideCloseIcon?: boolean;
+  } & styles.DialogVariants
+>;
+export const Dialog = forwardRef<HTMLDivElement, DialogProps>(
+  ({ children, hideCloseIcon = false, size = "md", ...props }, ref) => {
+    return (
+      <Box asChild ref={ref} {...props}>
+        <RadixDialog.Root>
+          <Box asChild>
+            <RadixDialog.Portal>
+              <Flex asChild {...styles.overlay()}>
+                <RadixDialog.Overlay>
```

Comment:
this structure doesn't seem right - box shouldn't be inside the overlay

overlay and box should be siblings

```tsx
<Portal>
  <Overlay />

  <Content />
</Portal>
```

Reply:
also, let's invert the pattern for composing box with radix - so instead of having the flex outside let's have it inside

```suggestion
              <RadixDialog.Overlay asChild>
                <Flex {...styles.overlay()}>
```